server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -930,6 +931,50 @@ func (s *BuildBuddyServer) getAPIKeysForAuthorizedGroup(ctx context.Context) ([]
 	return toProtoAPIKeys(append(userKeys, groupKeys...)), nil
 }
 
+func (s *BuildBuddyServer) replaceURLSubdomain(ctx context.Context, rawURL string) string {
+	if !subdomain.Enabled() {
+		return rawURL
+	}
+
+	u, err := perms.AuthenticatedUser(ctx, s.env)
+	if err != nil {
+		return rawURL
+	}
+
+	g, err := s.env.GetUserDB().GetGroupByID(ctx, u.GetGroupID())
+	if err != nil {
+		return rawURL
+	}
+
+	if g.URLIdentifier == nil || *g.URLIdentifier == "" {
+		return rawURL
+	}
+
+	pURL, err := url.Parse(rawURL)
+	// Input URLs should already be validated so this is not supposed to happen.
+	if err != nil {
+		alert.UnexpectedEvent("invalid_config_url", rawURL)
+		return rawURL
+	}
+
+	hostname := pURL.Hostname()
+	pts := strings.Split(hostname, ".")
+	var domain string
+	if len(pts) < 2 {
+		domain = hostname
+	} else {
+		domain = strings.Join(pts[len(pts)-2:], ".")
+	}
```

Comment:
might be nice to add a ParseDomain() function in utils somewhere for this kind of thing. Didn't we have very similar code somewhere else as well?